ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -60,6 +62,10 @@
  * will be accepted and all scheduled writes will be notified of the failure.
  */
 class NettyResponseChannel implements RestResponseChannel {
+  // add to this list if the connection needs to be closed on certain errors on GET, DELETE and HEAD.
+  // for a POST, we always close the connection on error because we expect the channel to be in a bad state.
+  static final List<HttpResponseStatus> CLOSE_CONNECTION_ERROR_STATUSES = new ArrayList<>();
```

Comment:
Yes that is correct. You have it right but I just want to make it clear for others (and also to make sure that my logic/assumptions are correct).

On POST, the reason to not keep it open is because we the data stream from the client might still be coming in and we might declare an error. This means that the connection is "dirty".

On any other method, there is no content so we know that the request is fully in. And we know that when we respond with an error, the error is fully formed. This means that the connection is "clean" and can be reused. 

Note that connections are still closed if there is an error after the response was sent and while get content is still being streamed to the client.

As for your question about the client, I did not understand it. The client always sends a keep-alive header and if the server says "Connection: close" in the response, then it closes the connection (or if it detects an error or closing from the other end). Otherwise it keeps it open.